Networking/Networking/Model/PaymentGatewayAccount.swift

```diff
@@ -30,6 +30,14 @@ public struct PaymentGatewayAccount: GeneratedCopiable, GeneratedFakeable {
     /// A boolean flag indicating if this Account is eligible for card present payments
     public let isCardPresentEligible: Bool
 
+    /// Indicates if the account is live (i.e. can accept actual payments)
+    public let isLive: Bool
+
+    /// Indicates if the gateway is set for test mode. This is NOT the same as
+    /// whether the account is live or not. You can have a live account set for
+    /// test mode, although we cannot accept in-person payments in that situation.
+    public let isInTestMode: Bool
```

Comment:
❓ I think I'm right in understanding that `testMode` is a property of the site, not the account... with that in mind, is there anywhere more semantically correct that this property might live? If the answer's no, that's fine!

Reply:
`testMode` is not a site property but a property of some payment gateways, like WooCommerce Payments. I'd daresay most payment gateways have a "test" or "debug" mode that the merchant can enable in settings for that gateway.

The major downside is there is no standard as to what that test mode / debug mode setting is called. So over the years developers that have added test modes to their gateways have used all sorts of things.

If there had been a standard, we could have more easily added the concept of test mode to `PaymentGateway` - which has a REST API at `wc/v3/payment_gateways` and specifically for WCPay at `wc/v3/payment_gateways/woocommerce_payments` 

But without that, we'd could probably have to entertain saving a whitelist of fields of various names (starting with the test_mode setting in WCPay) from `settings` portion for each gateway in the response to its `PaymentGateway` model, and then use that instead of getting test mode from the WCPay plugins' account response as we are doing here.

We'd still want this "associated account" model (PaymentGatewayAccount) for those concepts that are related to the backend account connected to a payment gateway (e.g. is a live account, has restrictions coming from the processor, etc) but we could at least stop relying on test_mode from here if we moved it to `PaymentGateway`

That's all assuming a shop manager can access `wc/v3/payment_gateways/woocommerce_payments` of course. I haven't checked that.

How about we start a refactor proposal post or issue for this? cc @ctarda

Reply:
Thanks for the explanation, that makes a lot more sense... I'd not really twigged that this was an account for the gateway, rather than the. I should have, given the name, but I blame Friday brain!

That refactor would be nice but sounds like it might be pretty low priority